tests/unit/types/argument-set-test.js

```diff
@@ -35,53 +35,53 @@ test("it will filter out Arguments with an undefined value", function(assert) {
   let set = new ArgumentSet();
   set.push(new Argument("status", undefined));
 
-  assert.equal(set.length, 0);
+  assert.equal(set.toArray().length, 0);
 });
 
 test("it will not filter out Arguments with a null value", function(assert) {
   let set = new ArgumentSet();
   set.push(new Argument("status", null));
 
-  assert.equal(set.length, 1);
+  assert.equal(set.toArray().length, 1);
 });
 
 
 test("it can be made from a query", function(assert) {
   let set = ArgumentSet.fromQuery({ status: 'active', limit: 10 });
 
-  assert.equal(set.length, 2);
-  assert.equal(set[0].name, 'status');
-  assert.equal(set[0].value, 'active');
-  assert.equal(set[1].name, 'limit');
-  assert.equal(set[1].value, 10);
+  assert.equal(set.toArray().length, 2);
+  assert.equal(set.toArray()[0].name, 'status');
```

Comment:
Nit: it looks like the most common operation is `length` and index `[]` so maybe you could create methods on the set to avoid having to call `toArray`. For example

```
function length() {
   toArray().length
}

function get(index) {
  toArray()[index];
}
```
That way you save having to type `toArray` every time. 

Up to you though.